engine/src/main/java/org/destinationsol/game/ColorSpan.java

```diff
@@ -54,11 +54,11 @@ public RgbSpan(Color start, Color end) {
 
         @Override
         public void set(float perc, Color col) {
-            perc = MathUtils.clamp(perc, (float) 0, (float) 1);
-            col.r = midVal(myStart.r, myEnd.r, perc);
-            col.g = midVal(myStart.g, myEnd.g, perc);
-            col.b = midVal(myStart.b, myEnd.b, perc);
-            col.a = midVal(myStart.a, myEnd.a, perc);
+            float percentage = MathUtils.clamp(perc, (float) 0, (float) 1);
```

Comment:
As stated above, to make it clear to the caller and to guide against bugs `perc` can be annotated as `final float perc` in the parameter list.
This is in addition to introduce a local variable.